GOVERNANCE.md

```diff
@@ -0,0 +1,458 @@
+# Governance thin-edge.io
+
+This documents serves as a guiding document to ease technical decision-making in the project as a whole.
+
+Some of its goals are:
+
+- Making sure that the different stakeholders are represented with their requirements
+- Making sure that contributors know how decisions are made and who they can address concerns/questions to
+- Making sure that team members know what is required of them for the project
+
+
+## The project vision
+
+In the [vision.md](./vision.md) the goal of the project is encoded.
+It specifies the different functional requirements and non-functional requirements of the project as a whole.
+All the stakeholders must agree on the vision upon joining.
+
+- The current stakeholders are: Software AG, and IFM
+
+Changes to the vision need to be done in a plenum of stakeholders, this makes sure that everyone is aware and agrees to the evolution of the project.
+
+
+
+## Team Structure
+
+* Teams are either single contributors or a group of them
+* They are responsible for a specific part of project, and as such are the maintainers of those parts
+    * This includes: Reviewing pull requests, triaging issues in their assigned area, and general maintenance
+* Each team is assigned a single or multiple subtrees of the project, as defined by the project structure
+* The teams follow the project hierarchy for decisions, i.e. higher decisions take precedence
+
+The important bits:
+
+At the 'top' are the maintainers, whose job is to define and realize the project vision of thin-edge.io.
+Maintainers should strive for agreement based on the following factors:
+
+- The vision of the whole project
+- Feasibility, accounting for future technical and cultural shifts
+- The health of the project going forward
```

Comment:
While the current bullets are fine, I think there is need for some "value" to the user aspect. e.g one can build great framework which might be feasible and align to braoder vision but not add serious value to the users.

Reply:
@itsroshan  agree here , also commented on a similar line below, we need to work against common goals / success metrics and user adoption (click rates, forks, pull requests, issues , comments, number of community event attendees, discord users are all metrics we should include in that common goal) we can keep it generic in this suggestion and add a explanation of what we consider as "healthy" e.g. liunk to the vision doc and metrics

Reply:
@itsroshan 

I agree that putting in a general 'value' statement into the vision can be a great idea. There are already some points, but they can be expanded of course. (Patch suggestions welcome ;) )


@AndrejS321 

w.r.t. 'metrics': I won't stop you from doing your own analysis, but I do not agree that any meaningful data can be derived from interactions. If anything, the juicy metrics are hidden away in questions like "How easy did you find it to use?" and click rates won't help you answer that.

I find it also worrisome to try and include this in the public-facing governance, as users are here to have a problem solved, and not tracked and analyzed.

Since both Software AG and IFM will be consumers of this framework, I consider that sufficient quality assurances in a first step. Once others find this framework useful, they will hopefully voice their concerns (if any).

Reply:
@TheNeikos  I agree, metrics do not need to be in the governance doc (and could even be tracked "internally"), but we need to decide somewhere what "healthy" for us means. Perceived "quality" is not enough from my point of view, one of the key objective of the maintainers is to make sure the project is "healthy" in terms of adoption, and this should also make it easier to decide on architecture topics.  e.g if adoption is happening from outside of SAG and ifm, this would be very positive, and I would apply it to the core framework and plug-ins aswell. 

"I find it also worrisome to try and include this in the public-facing governance, as users are here to have a problem solved, and not tracked and analyzed." --> my statement was not specific enough, I was referring to the metrics which GitHub and other community channels already give us to measure if we are on the right way with user adoption.

Reply:
Conclusion: 
- the "value and health" of the project will be defined in the vision docs (which will serve as a "alignment" mechanism between all maintainers/contributors, e.g. defining common goals)